packages/netsuite-adapter/src/change_validator.ts

```diff
@@ -146,16 +150,24 @@ const getChangeValidator: ({
       additionalDependencies,
       filtersRunner,
       elementsSource,
+      validatorsConfig,
     }
   ) =>
     async (changes, elementSource) => {
-      const validators = withSuiteApp
-        ? [...netsuiteChangeValidators, ...onlySuiteAppValidators]
-        : [...netsuiteChangeValidators, ...nonSuiteAppValidators]
-
+      const netsuiteValidators = withSuiteApp
+        ? { ...netsuiteChangeValidators, ...onlySuiteAppValidators }
+        : { ...netsuiteChangeValidators, ...nonSuiteAppValidators }
+
+      // Converts NetsuiteChangeValidator to ChangeValidator
+      const validators: Record<string, ChangeValidator> = _.mapValues(netsuiteValidators, validator =>
+        (innerChanges: ReadonlyArray<Change>) => validator(innerChanges, deployReferencedElements, elementsSource))
+
+      const mergedValidator = createChangeValidator({
+        validators: { ...defaultChangeValidators, ...validators },
+        validatorsConfig,
+      })
       const validatorChangeErrors: ChangeError[] = _.flatten(await Promise.all([
-        ...changeValidators.map(validator => validator(changes, elementSource)),
-        ...validators.map(validator => validator(changes, deployReferencedElements, elementsSource)),
+        mergedValidator(changes, elementSource),
         warnStaleData ? safeDeployValidator(changes, fetchByQuery, deployReferencedElements) : [],
       ]))
```

Comment:
can you do this instead?
```suggestion
      const safeDeploy = warnStaleData
        ? { safeDeploy: (innerChanges: ReadonlyArray<Change>) =>
          safeDeployValidator(innerChanges, fetchByQuery, deployReferencedElements) }
        : {}
      const mergedValidator = createChangeValidator({
        validators: { ...defaultChangeValidators, ...validators, ...safeDeploy },
        validatorsConfig,
      })
      const validatorChangeErrors = await mergedValidator(changes, elementSource)
```